akka-remote/src/main/scala/akka/remote/PhiAccrualFailureDetector.scala

```diff
@@ -122,8 +134,11 @@ class PhiAccrualFailureDetector(
         // this is a known connection
         val interval = timestamp - latestTimestamp
         // don't use the first heartbeat after failure for the history, since a long pause will skew the stats
-        if (isAvailable(timestamp)) oldState.history :+ interval
-        else oldState.history
+        if (isAvailable(timestamp)) {
+          if (interval >= (acceptableHeartbeatPauseMillis / 2) && eventStream.isDefined)
+            eventStream.get.publish(Warning(this.toString, getClass, s"heartbeat interval is growing too large: $interval millis"))
```

Comment:
ok, there is no `toString` implementation here so it will be the class name + hashCode. That can be good in this case. In general I think we have been rather inconsistent with those logSource strings making it difficult to filter/grep logs.